### DISCUSSION TITLE
Running the Oil Spill. I encountered this error

### DISCUSSION THREAD

**Discussion by joecabz1985** [OpenDrift/opendrift#1174] [Q&A]
21:55:12 INFO    opendrift.models.basemodel:2819: ========================
Traceback (most recent call last):
  File "/home/sirjoecabz/miniconda3/opendrift/opendrift/models/basemodel/__init__.py", line 2777, in run
    raise ValueError(
ValueError: No more active or scheduled elements, quitting.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/sirjoecabz/miniconda3/opendrift/MySimulation/phil_openoil.py", line 35, in <module>
    o.run(time_step=timedelta(minutes=15), time_step_output=timedelta(hours=3),
  File "/home/sirjoecabz/miniconda3/opendrift/opendrift/models/basemodel/__init__.py", line 2823, in run
    raise ValueError('Simulation stopped within '
ValueError: Simulation stopped within first timestep. "Missing variables: ['x_sea_water_velocity', 'y_sea_water_velocity']", 'The simulation stopped before requested end time was reached.'

**Comment by joecabz1985** [comment]
#!/usr/bin/env python
"""
Openoil
==================================
"""

from datetime import datetime, timedelta
from opendrift.models.openoil import OpenOil

o = OpenOil()

#print(o.oiltypes)  # Print available oil types

#%% Add forcing date
#Add atmospheric and ocean model
o.add_readers_from_list([
    'https://nrt.cmems-du.eu/thredds/dodsC/cmems_mod_glo_phy_anfc_merged-uv_PT1H-i',
    #'https://tds.hycom.org/thredds/dodsC/GLBy0.08/expt_93.0/uv3z',
    'https://pae-paha.pacioos.hawaii.edu/thredds/dodsC/ncep_global/NCEP_Global_Atmospheric_Model_best.ncd'])
#%%
# Seeding some particles
o.seed_elements(time=datetime(2023, 2, 28,12), lon=121.6, lat=13.35, z=360,
                number=80000, oil_type='GENERIC DIESEL')

#%%
# Adjusting some configuration
o.set_config('processes:evaporation',  True)
o.set_config('processes:emulsification',  True)
o.set_config('drift:vertical_mixing',  True)
o.set_config('vertical_mixing:timestep',  5)
o.set_config('drift:horizontal_diffusivity', 10)  # m2/s

#%%
# Running model
o.run(time_step=timedelta(minutes=15), time_step_output=timedelta(hours=3),
      end_time=datetime(2023, 3, 23, 12), outfile='openoilspill.nc')
#o.run(time_step=timedelta(minutes=15), duration=timedelta(days=20))
This is my code
#%%
# Print and plot results
print(o)
o.plot(fast=True)
o.plot_oil_budget()

#o.plot(filename='openoil_drift')
o.plot_vertical_distribution(maxnum=100,bins=50)
o.plot_property('water_fraction', mean=True)
o.plot_property('z')
o.plot_property('mass_evaporated')
o.plot_property('water_fraction')
o.plot_property('interfacial_area')
o.plot(linecolor='z', buffer=.1, show_elements=False, fast=False)
o.animation(color='z', buffer=.1 fast=True)
o.animation(fast=True, clabel='Ocean current [m/s]', background=['x_sea_water_velocity', 'y_sea_water_velocity'],filename='openoilspill_final.gif')

#%%
# .. image:: /gallery/animations/example_openoil_0.gif

**Comment by jehacode** [comment]
Hey,
make sure that the x_sea_water_velocity and y_sea_water_velocity are available you can check that by printing the reader,

another solution is check the logs 
ex
14:54:46 INFO    opendrift.readers.basereader:160: Variable y_sea_water_velocity will be rotated from northward_sea_water_velocity
14:54:46 INFO    opendrift.readers.basereader:160: Variable x_sea_water_velocity will be rotated from eastward_sea_water_velocity

and replace it with related field
o.animation(fast=True, clabel='Ocean current [m/s]', background=['eastward_sea_water_velocity', 'northward_sea_water_velocity'],filename='openoilspill_final.gif')
